src/libmongoc/src/mongoc/mongoc-crypt.c

```diff
@@ -696,6 +790,27 @@ _state_need_kms_credentials (_state_machine_t *sm, bson_error_t *error)
    const bool callback_provided_aws =
       bson_iter_init_find (&iter, &creds, "aws");
 
+   // Whether the callback requested auto-Azure credentials
+   const bool cb_wants_auto_azure = _check_azure_kms_auto (&creds, error);
+   if (error->code) {
+      // _check_azure_kms_auto failed
+      goto fail;
+   }
+   // Whether the original kmsProviders requested auto-Azure credentials
+   const bool orig_wants_auto_azure =
+      !cb_wants_auto_azure &&
+      _check_azure_kms_auto (&sm->crypt->kms_providers, error);
+   if (error->code) {
+      // _check_azure_kms_auto failed
+      goto fail;
+   }
+   const bool wants_auto_azure = cb_wants_auto_azure || orig_wants_auto_azure;
```

Comment:
Checking `cb_wants_auto_azure` differs from the `aws` logic below. If the callback provides `"aws": {}` that will not attempt to add AWS credentials from the environment.

I do not see a clear advantage to either approach. Consider making the logic consistent, or adding a comment to note the inconsistency.

Reply:
The callback function in this situation is unfortunately under-specified. I think it will be better to go with the AWS definition for now.